lib/internal/test_runner/runner.js

```diff
@@ -801,9 +800,23 @@ function run(options = kEmptyObject) {
     }
   }
 
-  const setupPromise = PromiseResolve(setup?.(root.reporter));
-  PromisePrototypeThen(PromisePrototypeThen(PromisePrototypeThen(setupPromise, runFiles), postRun), teardown);
+  const runChain = async () => {
+    if (typeof setup === 'function') {
+      await setup(root.reporter);
+    }
```

Comment:
```suggestion
    await setup?.(root.reporter);
```

Reply:
I would rather not make this change because I only want to `await` the value if it is actually a `Promise`.

Reply:
Is there any downside in awaiting `undefined`?

Reply:
My understanding is that it will still introduce unnecessary turns of the event loop.